src/upnp.cpp

```diff
@@ -708,6 +708,22 @@ void upnp::post(upnp::rootdevice const& d, char const* soap
 #endif
 }
 
+std::string upnp::create_soap(char const* soap_action, std::string const& service_namespace, char const* part)
```

Comment:
for example, take this as a parameter: ``std::initializer_list<std::pair<string_view, string_view>> args``.

and build the string like something like this (in a loop):

```
span<char> buf = buffer;
for (auto a : args)
{
   int written = std::snprintf(buf.data(), buf.size(), "<%*s>%*s</%*s>\n"
      , int(a.first.size()), a.first.data()
      , int(a.second.size()), a.second.data()
      , int(a.first.size()), a.first.data());
   buf = buf.subspan(written);
}
```

something like that. Then you can call it and pass in: ``{ {"NewLeaseDuration"_sv, string_view(duration), { ... }} }``.

For the string view literal, use this: https://github.com/arvidn/libtorrent/blob/master/include/libtorrent/string_view.hpp#L60